plugins/reports/counter/classes/LegacyJR1.inc.php

```diff
@@ -0,0 +1,358 @@
+
+		return array($begin, $end);
+	}
+
+	/**
+	 * Get the valid journal IDs for which log entries exist in the DB.
+	 * @param $useLegacyStats boolean Use the old counter plugin data.
+	 * @return array
+	 */
+	function _getJournalIds($useLegacyStats = false) {
+		$metricsDao =& DAORegistry::getDAO('MetricsDAO'); /* @var $metricsDao MetricsDAO */
+		if ($useLegacyStats) {
+			$results = $metricsDao->getMetrics(OJS_METRIC_TYPE_LEGACY_COUNTER, array(STATISTICS_DIMENSION_ASSOC_ID));
+			$fieldId = STATISTICS_DIMENSION_ASSOC_ID;
+		} else {
+			$filter = array(STATISTICS_DIMENSION_ASSOC_TYPE => ASSOC_TYPE_GALLEY);
+			$results = $metricsDao->getMetrics(OJS_METRIC_TYPE_COUNTER, array(STATISTICS_DIMENSION_CONTEXT_ID), $filter);
+			$fieldId = STATISTICS_DIMENSION_CONTEXT_ID;
+		}
+		$journalIds = array();
+		foreach($results as $record) {
+			$journalIds[] = $record[$fieldId];
+		}
+		return $journalIds;
+	}
+
+	/**
+	 * Retrieve a monthly log entry range.
+	 * @param $journalId int
+	 * @param $begin
+	 * @param $end
+	 * @param $useLegacyStats boolean Use the old counter plugin data.
+	 * @return 2D array
+	 */
+	function _getMonthlyLogRange($journalId = null, $begin, $end, $useLegacyStats = false) {
+		$begin = date('Ym', strtotime($begin));
+		$end = date('Ym', strtotime($end));
+
+		$metricsDao =& DAORegistry::getDAO('MetricsDAO'); /* @var $metricsDao MetricsDAO */
+		$columns = array(STATISTICS_DIMENSION_MONTH, STATISTICS_DIMENSION_FILE_TYPE);
+		$filter = array(
+			STATISTICS_DIMENSION_MONTH => array('from' => $begin, 'to' => $end)
+		);
+
+		if ($useLegacyStats) {
+			$dimension = STATISTICS_DIMENSION_ASSOC_ID;
+			$metricType = OJS_METRIC_TYPE_LEGACY_COUNTER;
+		} else {
+			$dimension = STATISTICS_DIMENSION_CONTEXT_ID;
+			$metricType = OJS_METRIC_TYPE_COUNTER;
+			$filter[STATISTICS_DIMENSION_ASSOC_TYPE] = ASSOC_TYPE_GALLEY;
+		}
+
+		if ($journalId) {
+			$columns[] = $dimension;
+			$filter[$dimension] = $journalId;
+		}
+
+		$results = $metricsDao->getMetrics($metricType, $columns, $filter);
+		return $results;
+	}
+
+	/**
+	 * Retrieve a monthly log entry range.
+	 * @param $begin
+	 * @param $end
+	 * @param $useLegacyStats boolean Use the old counter plugin data.
+	 * @return 2D array
+	 */
+	function _getMonthlyTotalRange($begin, $end, $useLegacyStats = false) {
+		return $this->_getMonthlyLogRange(null, $begin, $end, $useLegacyStats);
+	}
+
+	/**
+	 *
+	 * Counter report in XML
+	 * @param $request PKPRequest
+	 * @param $year string
+	 * @param $useLegacyStats boolean
+	 */
+	function _reportXML(&$request, $year, $useLegacyStats) {
+		$templateManager =& TemplateManager::getManager();
+		list($begin, $end) = $this->_getLimitDates($year);
+
+		$this->_assignTemplateCounterXML($templateManager, $begin, $end, $useLegacyStats);
+		$templateManager->display($this->getParentPlugin()->getTemplatePath() . 'reportxml.tpl', 'text/xml');
+	}
+
+	/**
+	 * Get the parent plugin
+	 * @return object
+	 */
+	function &getParentPlugin() {
```

Comment:
I'll pass in the template path here.

The legacy report does not implement CounterReport because the core idea behind CounterReport is to transform a metrics request into a set of COUNTER\ReportItems.  These ReportItems can then be represented as XML.  COUNTER\ReportItems are now tied to the current COUNTER Release.  I had intended originally to support multiple COUNTER releases in the COUNTER classes and in the plugin, but this is unnecessary because COUNTER itself does not support old releases.  Rather, they give a deadline for all implementations to support the latest release: http://www.projectcounter.org/code_practice.html